sale_commission/models/sale_commission_mixin.py

```diff
@@ -10,32 +10,24 @@ class SaleCommissionMixin(models.AbstractModel):
         "Mixin model for applying to any object that wants to " "handle commissions"
     )
 
-    @api.model
     def _prepare_agents_vals_partner(self, partner):
         """Utility method for getting agents of a partner."""
-        rec = []
-        for agent in partner.agents:
-            rec.append((0, 0, {"agent": agent.id, "commission": agent.commission.id,}))
-        return rec
-
-    @api.model
-    def _default_agents(self):
-        agents = []
-        context = self.env.context
-        if context.get("partner_id"):
-            partner = self.env["res.partner"].browse(context["partner_id"])
-            agents = self._prepare_agents_vals_partner(partner)
-        return agents
-
-    agents = fields.One2many(
+        return [
+            (0, 0, {"agent": agent.id, "commission": agent.commission.id})
+            for agent in partner.agent_ids
+        ]
+
+    # TODO: Rename it to agent_ids
```

Comment:
Already done :)

Reply:
And with migration scripts?

Reply:
I don't know why the commission detail does not appear in that view, I am monitoring the creation of the record in the table "sale_order_line_agent" and it does it correctly.

The recalculation of commissions has already been corrected.